p2p/gen/class.go

```diff
@@ -0,0 +1,12 @@
+package gen
```

Comment:
This package is an adapter to create aliases for types in new specs with so the call site doesn't change. Next step would be removing this package and use the new types directly.

Reply:
Can we add a todo in the code or an issue that reflect this intention, unless it's going to be part of an immediate follow up PR

Reply:
#2824 will be the follow-up PR, so I won't add a TODO here.